Pyroscope datasource: Crash when entering query

### What happened?

I am running a build from main (commit `b08271da32`). When I enter any query in the Pyroscope datasource I get a crash like this:

```
panic: interface conversion: interface {} is int64, not data.EnumItemIndex                                              
                                                                                                                        
goroutine 38019 [running]:                                                                                              
github.com/grafana/grafana-plugin-sdk-go/data.(*enumVector).Append(0x52e1340?, {0x4fbede0?, 0x9db76f8?})                
»·······/opt/drone/gomodcache/github.com/grafana/grafana-plugin-sdk-go@v0.171.0/data/field_type_enum.go:32 +0xb7        
github.com/grafana/grafana-plugin-sdk-go/data.(*Field).Append(...)                                                      
»·······/opt/drone/gomodcache/github.com/grafana/grafana-plugin-sdk-go@v0.171.0/data/field.go:160                       
github.com/grafana/grafana/pkg/tsdb/grafana-pyroscope-datasource.(*EnumField).Append(0xc008cc9d48, {0xc005bcbbc0, 0x5}) 
»·······/drone/src/pkg/tsdb/grafana-pyroscope-datasource/query.go:328 +0x82                                             
github.com/grafana/grafana/pkg/tsdb/grafana-pyroscope-datasource.treeToNestedSetDataFrame.func1(0xc003d2c7d0)           
»·······/drone/src/pkg/tsdb/grafana-pyroscope-datasource/query.go:305 +0xea                                             
github.com/grafana/grafana/pkg/tsdb/grafana-pyroscope-datasource.walkTree(0xc002c822d0, 0xc008cc9d60)                   
»·······/drone/src/pkg/tsdb/grafana-pyroscope-datasource/query.go:362 +0x6e                                             
github.com/grafana/grafana/pkg/tsdb/grafana-pyroscope-datasource.treeToNestedSetDataFrame(0xc002c822d0, {0x5e37f32, 0x2})
»·······/drone/src/pkg/tsdb/grafana-pyroscope-datasource/query.go:301 +0x3aa                                            
github.com/grafana/grafana/pkg/tsdb/grafana-pyroscope-datasource.responseToDataFrames(0xc003077d58)                     
»·······/drone/src/pkg/tsdb/grafana-pyroscope-datasource/query.go:126 +0x4d                                             
github.com/grafana/grafana/pkg/tsdb/grafana-pyroscope-datasource.(*PhlareDatasource).query.func2()                      
»·······/drone/src/pkg/tsdb/grafana-pyroscope-datasource/query.go:94 +0x2c7                                             
golang.org/x/sync/errgroup.(*Group).Go.func1()                                                                          
»·······/opt/drone/gomodcache/golang.org/x/sync@v0.3.0/errgroup/errgroup.go:75 +0x64                                    
created by golang.org/x/sync/errgroup.(*Group).Go                                                                       
»·······/opt/drone/gomodcache/golang.org/x/sync@v0.3.0/errgroup/errgroup.go:72 +0xa5                                    
```

### What did you expect to happen?

Query works properly and results are returned

### Did this work before?

I've got another deployment alongside the CD one running `v10.0.2` and that one works fine

### How do we reproduce it?

1. Run main, have a Pyroscope datasource configured
2. Visit Explore and issue a query against the datasource, e.g. `{cluster="mycluster", job="myjob"}

### Is the bug inside a dashboard panel?

_No response_

### Environment (with versions)?

Grafana: b08271da32
OS: Linux
Browser: Chrome linux 114.0.5735.198 (Official Build) unknown (64-bit) 

### Grafana platform?

Kubernetes

### Datasource(s)?

pyroscope

--- a/pkg/tsdb/grafana-pyroscope-datasource/query.go
+++ b/pkg/tsdb/grafana-pyroscope-datasource/query.go
@@ -312,14 +312,14 @@ func treeToNestedSetDataFrame(tree *ProfileTree, unit string) *data.Frame {
 
 type EnumField struct {
 	field     *data.Field
-	valuesMap map[string]int64
-	counter   int64
+	valuesMap map[string]data.EnumItemIndex
+	counter   data.EnumItemIndex
 }
 
 func NewEnumField(name string, labels data.Labels) *EnumField {
 	return &EnumField{
 		field:     data.NewField(name, labels, []data.EnumItemIndex{}),
-		valuesMap: make(map[string]int64),
+		valuesMap: make(map[string]data.EnumItemIndex),
 	}
 }
 
@@ -328,7 +328,7 @@ func (e *EnumField) Append(value string) {
 		e.field.Append(valueIndex)
 	} else {
 		e.valuesMap[value] = e.counter
-		e.field.Append(data.EnumItemIndex(e.counter))
+		e.field.Append(e.counter)
 		e.counter++
 	}
 }

--- a/pkg/tsdb/grafana-pyroscope-datasource/query_test.go
+++ b/pkg/tsdb/grafana-pyroscope-datasource/query_test.go
@@ -196,6 +196,7 @@ func Test_treeToNestedDataFrame(t *testing.T) {
 				},
 				{Value: 30, Level: 1, Self: 3, Name: "func2", Nodes: []*ProfileTree{
 					{Value: 15, Level: 2, Self: 4, Name: "func1:func3"},
+					{Value: 10, Level: 2, Self: 4, Name: "func1"},
 				}},
 			},
 		}
@@ -211,10 +212,10 @@ func Test_treeToNestedDataFrame(t *testing.T) {
 		}
 		require.Equal(t,
 			[]*data.Field{
-				data.NewField("level", nil, []int64{0, 1, 1, 2}),
-				data.NewField("value", nil, []int64{100, 40, 30, 15}).SetConfig(&data.FieldConfig{Unit: "short"}),
-				data.NewField("self", nil, []int64{1, 2, 3, 4}).SetConfig(&data.FieldConfig{Unit: "short"}),
-				data.NewField("label", nil, []data.EnumItemIndex{0, 1, 2, 3}).SetConfig(labelConfig),
+				data.NewField("level", nil, []int64{0, 1, 1, 2, 2}),
+				data.NewField("value", nil, []int64{100, 40, 30, 15, 10}).SetConfig(&data.FieldConfig{Unit: "short"}),
+				data.NewField("self", nil, []int64{1, 2, 3, 4, 4}).SetConfig(&data.FieldConfig{Unit: "short"}),
+				data.NewField("label", nil, []data.EnumItemIndex{0, 1, 2, 3, 1}).SetConfig(labelConfig),
 			}, frame.Fields)
 	})
 

